### ISSUE TITLE
not adding QUERY_ALL_PACKAGES to plugin itself

### ISSUE THREAD

**Issue by daadu** [g123k/flutter_plugin_device_apps#53]
[Android docs on package visibility](https://developer.android.com/training/basics/intents/package-visibility#all-apps) suggests that `QUERY_ALL_PACKAGES` should only be used with certain kind of apps.

> The following list gives some examples of use cases where the QUERY_ALL_PACKAGES permission is appropriate to include:
> 
> -  Launcher apps
> - Accessibility apps
> - Browsers
> - Peer-to-peer (P2P) sharing apps
> - Device management apps
> - Security apps

Instead we should ask user to read the docs themselves and leave it to themselves to add (in spirit of "minimum privileges" !) - if to query all packages or specific packages with `<queries>` entry in manifest (of course, we need to write TLDR version of the android doc in README to help user make this decision).

In my case, I just want to check if a particular app is installed in device, I don't want to risk app suspension when the policy is updated.

**Comment by daadu** [comment]
It is impossible (without forking) to override `QUERY_ALL_PACKAGES` (specified in plugin itself) by the app developer, if wants to just query some apps.

**Comment by ThrowJojo** [comment]
I could be wrong but I think if you just add the following line to your app's manifest you can override the behavior:

```xml
<uses-permission
        android:name="android.permission.QUERY_ALL_PACKAGES"
        tools:node="remove"/>
```

That being said I haven't tried it with this package yet but I'm pretty sure it will work. Maybe something should be added to the README about it considering there will potentially be policy changes soon.

**Comment by daadu** [comment]
Thank you for pointing it out, will test it and let you know if it works.

Yes, I agree that this should be mentioned in README

**Comment by ThrowJojo** [comment]
I migrated one of my projects to null safety so I decided to try the override out after upgrading this package to the latest version and it worked, I checked by examining the release APK's manifest inside Android Studio.

@g123k Thanks for creating this package, it's great. Sorry for pinging you but I was just wondering if you'd be willing to accept a pull request if I added a note to the README about overriding the  `QUERY_ALL_PACKAGES` permission?
